optimum/neuron/models/training/qwen3/modeling_qwen3.py

```diff
@@ -0,0 +1,223 @@
+# coding=utf-8
+# Copyright 2025 The Qwen team, Alibaba Group and the HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch Qwen3 model, partly based on Llama model and on Transformers implementation."""
+
+from functools import partial
+from typing import Optional, Tuple
+
+import torch
+from torch import nn
+from transformers.modeling_flash_attention_utils import FlashAttentionKwargs
+from transformers.models.qwen3.configuration_qwen3 import Qwen3Config
+from transformers.processing_utils import Unpack
+from transformers.utils import logging
+
+from ....utils import is_neuronx_distributed_available, is_torch_xla_available
+from ..config import TrainingNeuronConfig
+from ..llama.modeling_llama import (
+    LlamaAttention,
+    LlamaDecoderLayer,
+    LlamaForCausalLM,
+    LlamaModel,
+    LlamaPreTrainedModel,
+    LlamaRMSNorm,
+    LlamaRotaryEmbedding,
+    apply_rotary_pos_emb,
+    eager_attention_forward,
+    repeat_kv,
+)
+from ..modeling_utils import ALL_ATTENTION_FUNCTIONS
+
+
+if is_torch_xla_available():
+    pass
+
+if is_neuronx_distributed_available():
+    from neuronx_distributed.parallel_layers.layers import (
+        ParallelEmbedding,
+    )
+    from neuronx_distributed.parallel_layers.parallel_state import get_tensor_model_parallel_size
+
+logger = logging.get_logger(__name__)
+
+
+def _init_normal(std, w):
+    return nn.init.normal_(w, mean=0.0, std=std)
+
+
+class Qwen3Attention(LlamaAttention):
+    def __init__(self, config: Qwen3Config, mp_config: TrainingNeuronConfig, layer_idx: int):
```

Comment:
`mp_config` was called liked that for "Model Parallelism config", do you think we should `mp_config -> trn_config` ?